builders/tslice.py

```diff
@@ -0,0 +1,190 @@
+        "aggregation_control": {
+           "variable_column_name": "time_range",
+           "groupby_attrs": ["time_range"],
+           "aggregations":[
+                           {
+                           "type": "union",
+                           "attribute_name": "time_range"
+                           },
+                           {
+                           "type": "join_existing",
+                           "attribute_name": "time_range",
+                           "options": {
+                                      "dim": "time",
+                                      "coords": "minimal",
+                                      "compat": "override"
+                                      }
+                           },
+           ] 
+        }                         
+    }
+    # add a section for each column
+    for col in cols:
+        c = {
+            "column_name": col,
+            "vocabulary": ""
+            }
+        d["attributes"].append(c)
+
+    od=OrderedDict(d.items())
+    #write the json file
+    with open(json_filepath,'w') as f:
+        json.dump(od,f,indent=4)
+
+
+def verify(input_yaml):
+    # verify that we're working with a dictionary
+    if not isinstance(input_yaml,dict):
+        print("ERROR: The experiment/dataset top level is not a dictionary. Make sure you follow the correct format.")
+        return False
+    for dataset in input_yaml.keys():
+        # check to see if there is a data_sources key for each dataset
+        if 'data_sources' not in input_yaml[dataset].keys():
+            print("ERROR: Each experiment/dataset must have the key 'data_sources'. Verify "+dataset+" contains this key.")
+            return False
+        # verify that we're working with a list at this level
+        if not isinstance(input_yaml[dataset]['data_sources'],list):
+            print("ERROR: The data_sources are not in a list.  Make sure you follow the correct format.")
+            return False
+        for stream_info in input_yaml[dataset]['data_sources']:
+            # check to make sure that there's a 'glob_string' key for each data_source
+            if 'glob_string' not in stream_info.keys():
+                print("ERROR: Each data_source must contain a 'glob_string' key.  Verify that all data_sources under "+dataset+" contain a 'glob_string' key.")
+                return False
+    return True
+
+
+def common_parser(filepath, local_attrs, glob_attrs):
+    """
+    gather the file parts for the df
+    """
+    basename = os.path.basename(filepath)
+    fileparts = {}
+    fileparts['path'] = filepath
+
+    try:
+        fileparts['variable'] = []
+        # open file
+        d = nc.Dataset(filepath,'r')
+        # find what the time (unlimited) dimension is
+        dims = list(dict(d.dimensions).keys())
+        look_for_unlim=[d.dimensions[dim].isunlimited() for dim in dims]
+        unlim=[i for i, x in enumerate(look_for_unlim) if x]
+        unlim_dim=dims[unlim[0]]
```

Comment:
This is going to fail when `look_for_unlim` is an empty list:

<img width="800" alt="Screen Shot 2020-04-21 at 5 21 31 PM" src="https://user-images.githubusercontent.com/13301940/79924251-97c8ce80-83f4-11ea-85f5-bb9d2a62dc26.png">

Reply:
@andersy005 good catch.  I'll look at catching this.

Reply:
@andersy005 I'm using this code to include only time variant fields in the variable list.  Should it include all variables instead of limiting what is included?

Reply:
> Should it include all variables instead of limiting what is included?

Thinking about this a bit, I don't really know whether we should exclude time invariant fields or not. I am going to let others chime in. Cc @kmpaul 

If we include only time variant fields in the variable, is there any guarantee that the unlimited dimensions are always specified in the netCDF file?

Reply:
1. Usually, time-invariant fields are "pseudo-coordinates"...or metadata fields.  But I don't think we can guarantee that is true for *all* data.  Maybe we should conform to `xarray`'s standard practice, which I think is to include everything in the data variables unless it can be unambiguously identified as a "coordinate".

2. I don't think you can guarantee that "unlimited" dimensions will even exist.  My understanding is that some data, for reasons I cannot remember right now, explicitly removes the "unlimited" dimension from all files.

Reply:
I'll implement @kmpaul 's suggestion.